src/main/java/uk/ac/cam/cl/dtg/segue/api/AdminFacade.java

```diff
@@ -569,7 +570,107 @@ public synchronized Response setUsersEmailVerificationStatusFailed(
+     * @param webhookPayload
+     *            - a list of user webhookPayload that need to be changed
+     * @return Success shown by returning an ok response
+     */
+    @POST
+    @Path("/users/unsubscription_notification/{providerToken}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response notifyExternalUnsubscriptionEvent(@Context final HttpServletRequest request,
+                                                @PathParam("providerToken") final String providerToken,
+                                                final List<Map<String, Object>> eventDetailsList) {
+        try {
+            final String expectedProviderToken = getProperties().getProperty(MAILJET_WEBHOOK_TOKEN);
+            if (!expectedProviderToken.equals(providerToken)) {
+                return SegueErrorResponse.getIncorrectRoleResponse();
+            }
 
+            List<UserPreference> userPreferencesToUpdate = Lists.newArrayList();
+
+            for (Map<String, Object> eventDetails: eventDetailsList) {
+                String recipientEmail = (String) eventDetails.get("email");
+                // TODO: this is very MailJet specific:
+                String mailjetListId = (String) eventDetails.get("mj_list_id");
+                EmailType unsubscribedEmailType = EmailType.NEWS_AND_UPDATES;
+                if (getProperties().getProperty(MAILJET_NEWS_LIST_ID).equals(mailjetListId)) {
+                    unsubscribedEmailType = EmailType.NEWS_AND_UPDATES;
+                } else if (getProperties().getProperty(MAILJET_EVENTS_LIST_ID).equals(mailjetListId)) {
+                    unsubscribedEmailType = EmailType.EVENTS;
+                } else {
+                    log.warn(String.format("User with email (%s) attempted to unsubscribe from unrecognised list (%s)!", recipientEmail, mailjetListId));
+                }
+                // Find and unsubscribe user:
+                if (recipientEmail != null && !recipientEmail.isEmpty()) {
+                    try {
+                        RegisteredUserDTO user = userManager.getUserDTOByEmail(recipientEmail);
+                        UserPreference preferenceToSave = new UserPreference(user.getId(), SegueUserPreferences.EMAIL_PREFERENCE.name(), unsubscribedEmailType.name(), false);
+                        userPreferencesToUpdate.add(preferenceToSave);
+                    } catch (NoUserException e) {
+                        log.warn(String.format("User with email (%s) attempted to unsubscribe, but no Isaac account found!", recipientEmail));
+                    }
+                }
+            }
+            userPreferenceManager.saveUserPreferences(userPreferencesToUpdate);
+            String remoteIpAddress = RequestIPExtractor.getClientIpAddr(request);
+            log.info(String.format("Request from (%s) unsubscribed %s emails from NEWS_AND_UPDATES emails.", remoteIpAddress, userPreferencesToUpdate.size()));
```

Comment:
I don't think I am worried. The only malice that would actually matter (legally) would be deliberately resubscribing people which this doesn't allow; I really don't think it matters if we don't email someone who might have liked our email. I guess this endpoint would allow denial of service since it is slow, and timing requests might leak account existence, but (unfortunately) there are much easier endpoints to abuse for that.

Reply:
If they were to work through and unsubscribe everyone then marketing might be a little annoyed! Worst case, we could unset it so that the modal reappears for them on next visit to the site.